test/tests/interfaces/pbs_sched_interface_test.py

```diff
@@ -152,12 +175,128 @@ def test_set_and_unset_sched_attributes(self):
+        self.server.expect(SCHED, {'scheduling': 'True'},
+                           id='TestCommonSched', max_attempts=10)
+
+    def test_set_sched_priv_log_duplicate_value(self):
+        """
+        Test setting of sched_priv and sched_log to a
+        value assigned to another scheduler
+        """
+        err_msg = "Another scheduler also has same "
+        err_msg += "value for its sched_priv directory"
+        try:
+            self.server.manager(MGR_CMD_SET, SCHED,
+                                {'sched_priv': '/var/spool/pbs/sched_priv'},
+                                runas=ROOT_USER, id='TestCommonSched')
+        except PbsManagerError as e:
+            self.assertTrue(err_msg in e.msg[0],
+                            "Error message is not expected")
+        err_msg = "Another scheduler also has same "
+        err_msg += "value for its sched_log directory"
+        try:
+            self.server.manager(MGR_CMD_SET, SCHED,
+                                {'sched_log': '/var/spool/pbs/sched_logs'},
+                                runas=ROOT_USER, id='TestCommonSched')
+        except PbsManagerError as e:
+            self.assertTrue(err_msg in e.msg[0],
+                            "Error message is not expected")
+
+    def test_set_default_sched_not_permitted(self):
+        """
+        Test setting partition on default scheduler
+        """
+        err_msg = "Operation is not permitted on default scheduler"
+        try:
+            self.server.manager(MGR_CMD_SET, SCHED,
+                                {'partition': 'P1'},
+                                runas=ROOT_USER)
+        except PbsManagerError as e:
+            self.assertTrue(err_msg in e.msg[0],
+                            "Error message is not expected")
+        try:
+            self.server.manager(MGR_CMD_SET, SCHED,
+                                {'sched_priv': '/var/spool/somedir'},
+                                runas=ROOT_USER)
+        except PbsManagerError as e:
+            self.assertTrue(err_msg in e.msg[0],
+                            "Error message is not expected")
+        try:
+            self.server.manager(MGR_CMD_SET, SCHED,
+                                {'sched_log': '/var/spool/somedir'},
+                                runas=ROOT_USER)
+        except PbsManagerError as e:
+            self.assertTrue(err_msg in e.msg[0],
+                            "Error message is not expected")
+
+    def test_sched_name_too_long(self):
+        """
+        Test creating a scheduler with name longer than 15 chars
+        """
+        try:
+            self.server.manager(MGR_CMD_CREATE, SCHED,
+                                runas=ROOT_USER, id="TestLongScheduler")
+        except PbsManagerError as e:
+            self.assertTrue("Scheduler name is too long" in e.msg[0],
+                            "Error message is not expected")
+
+    def test_set_default_sched_attrs(self):
```

Comment:
@visheshh is there a test to check if route queue gets the partition name set on it? if not, can you add one. It is just a suggestion, you may chose to ignore it.

Reply:
It is already tested while testing partition interface and added in pbs_partition.py

Reply:
Thanks @visheshh It's good that the test case exists. I could not find pbs_partition.py in the pull through. Neither it is present in the repo. Am I missing something?

Reply:
It is already checked in interfaces folder .